.github/workflows/_build_publish.yaml

```diff
@@ -0,0 +1,251 @@
+      - uses: actions/upload-artifact@5d5d22a31266ced268874388b861e4b58bb5c2f3 # v4.3.1
+        id: binary-artifacts
+        with:
+          name: ${{ inputs.BINARY_ARTIFACT_NAME }}
+          path: |
+            ./build/distributions/out/*.tar.gz
+            ./build/distributions/out/*.sha256
+            !./build/distributions/out/*.tar.gz.sha256
+      - name: publish binaries
+        env:
+          PULP_USERNAME: ${{ vars.PULP_USERNAME }}
+          PULP_PASSWORD: ${{ secrets.PULP_PASSWORD }}
+          CLOUDSMITH_API_KEY: ${{ secrets.CLOUDSMITH_API_KEY }}
+        run: |
+          make publish/pulp
+  build-images:
+    runs-on: ubuntu-latest
+    outputs:
+      BINARY_ARTIFACT_DIGEST_BASE64: ${{ steps.inspect-binary-output.outputs.binary_artifact_digest_base64 }}
+    strategy:
+      fail-fast: false
+      matrix:
+        image: ${{ fromJSON(inputs.images) }}
+    steps:
+      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+        with:
+          fetch-depth: 0
+      - name: Install dependencies for cross builds
+        if: ${{ fromJSON(inputs.FULL_MATRIX) }}
+        run: |
+          sudo apt-get update; sudo apt-get install -y qemu-user-static binfmt-support
+      - uses: actions/setup-go@0c52d547c9bc32b1aa3301fd7a9cb496313a4491 # v5.0.0
+        with:
+          go-version-file: go.mod
+          cache-dependency-path: |
+            .build-info
+            go.sum
+      - uses: actions/cache@13aacd865c20de90d75de3b17ebe84f7a17d57d2 # v4.0.0
+        with:
+          path: |
+            ${{ env.CI_TOOLS_DIR }}
+          key: ${{ runner.os }}-${{ runner.arch }}-devtools-${{ hashFiles('mk/dependencies/deps.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-${{ runner.arch }}-devtools
+      - id: image_meta
+        run: |
+          echo "Extracting image meta for ${{ matrix.image }}"
+          registry=$(make docker/info/registry)
+          tag=$(make build/info/version)
+          echo "tag=${tag}" >> $GITHUB_OUTPUT
+          echo "registry=${registry}" >> $GITHUB_OUTPUT
+          echo "image=${registry}/${{ matrix.image }}:${tag}" >> $GITHUB_OUTPUT
+          # Add matrix to .build/info to cache
+          make build/info/short > .build-info
+      - run: |
+          make images/${{ matrix.image }}
+      - run: |
+          make docker/save/${{ matrix.image }}
+      - name: Run container structure test
+        if: ${{ !contains(github.event.pull_request.labels.*.name, 'ci/skip-container-structure-test') && !contains(github.event.pull_request.labels.*.name, 'ci/skip-test') }}
+        run: |
+          make test/container-structure/${{ matrix.image }}
+      - uses: actions/upload-artifact@5d5d22a31266ced268874388b861e4b58bb5c2f3 # v4.3.1
+        id: image-artifacts
+        with:
+          name: images_${{ matrix.image }}
+          path: |
+            ./build/docker/*.tar
+      # TODO Do we need to scan images for each arch?
```

Comment:
@saisatishkarra do you know if we can scan only one of the arch or we need to scan both?